WordPress/Classes/ViewRelated/Voice/VoiceToContentView.swift

```diff
@@ -1,36 +1,73 @@
-        NavigationView {
-            contents
-                .toolbar {
-                    ToolbarItem(placement: .cancellationAction) {
-                        Button("Cancel", action: buttonCancelTapped)
-                    }
-                }
-        }
-        .tint(Color(uiColor: .brand))
+        contents
+            .onAppear(perform: viewModel.onViewAppeared)
+            .tint(Color(uiColor: .brand))
         // TODO: Add proper error handing
-        .alert(viewModel.errorMessage ?? "", isPresented: $viewModel.isShowingError, actions: {
-            Button("OK", action: buttonCancelTapped)
-        })
+            .alert(viewModel.errorMessage ?? "", isPresented: $viewModel.isShowingError, actions: {
+                Button("OK", action: buttonCancelTapped)
+            })
     }
 
     @ViewBuilder
     private var contents: some View {
-        switch viewModel.state {
-        case .welcome:
-            VoiceToContentWelcomeView(viewModel: viewModel)
-        case .recording:
-            VoiceToContentRecordingView(viewModel: viewModel)
-        case .processing:
-            VoiceToContenProcessingView(viewModel: viewModel)
+        VStack {
+            headerView
+
+            if case .failed(let message, let onRetry) = viewModel.loadingState {
+                VStack(spacing: 32) {
+                    Text(message)
+                        .lineLimit(3)
+                        .foregroundStyle(.secondary)
+                    DSButton(title: Strings.retry, style: .init(emphasis: .secondary, size: .medium), action: onRetry)
+                }
+                .frame(maxWidth: .infinity, maxHeight: .infinity)
+            } else {
+                switch viewModel.step {
+                case .welcome:
+                    VoiceToContentWelcomeView(viewModel: viewModel)
+                case .recording:
+                    VoiceToContentRecordingView(viewModel: viewModel)
+                case .processing:
+                    VoiceToContenProcessingView(viewModel: viewModel)
+                }
+            }
+        }
+        .padding(22)
+        .padding(.vertical, 4)
+    }
+
+    private var headerView: some View {
+        HStack(alignment: .top) {
+            VStack(alignment: .leading, spacing: 12) {
+                Text(viewModel.title)
+                    .font(.title3.weight(.bold))
+                HStack {
+                    Text(viewModel.subtitle)
+                        .font(.subheadline.monospacedDigit())
+                        .foregroundStyle(.secondary)
+                    if case .welcome = viewModel.step, case .loading = viewModel.loadingState {
+                        ProgressView()
+                            .tint(.secondary)
+                            .controlSize(.small)
+                            .padding(.leading, 2)
+                    }
+                }
+            }
+            Spacer()
+            Button(action: viewModel.buttonCancelTapped) {
```

Comment:
Tapping on the button doesn't dismiss the view

Reply:
Fixed